lib/p2p/Pool.ts

```diff
@@ -135,6 +137,25 @@ class Pool extends EventEmitter {
     this.connected = false;
   }
 
+  private verifyReachability = () => {
+    this.handshakeData.addresses!.forEach(async (address) => {
+      const externalAddress = `${address.host}:${address.port}`;
+      this.logger.debug(`Trying to connect to advertised address: ${externalAddress}`);
+      try {
+        const peer = Peer.fromOutbound(address, this.logger);
+        await peer.open(this.handshakeData, this.handshakeData.nodePubKey);
+        peer.close();
+      } catch (err) {
```

Comment:
One more thing actually. In rereading this code flow, I think we should try to avoid try/catch as part of our *expected* flow and reserve `catch` statements for things that are truly exceptions. In this case since the exception is what we expect, I think the intent of this code would be more clear if we used [assert.rejects()](https://nodejs.org/api/assert.html#assert_assert_rejects_block_error_message). So instead of the try/catch, try something like

```javascript
assert.rejects(peer.open(this.handshakeData, this.handshakeData.nodePubKey), errors.ATTEMPTED_CONNECTION_TO_SELF);
```

I think we can drop the peer.close() statement at the end since that should be handled inside the `open()` method.

Reply:
I cant' get your example working:
```
{ message: 'Cannot attempt connection to self', code: '2.4' }

/home/michael/Git/JavaScript/xud/dist/bootstrap.js:13
        throw err;
        ^
[object Object]
```
and the this error isn't catchable. The `assert.rejects()` doc does exists for Node.js [v10.9.0](https://nodejs.org/api/assert.html#assert_assert_rejects_block_error_message) but not for [v8.11.4](https://nodejs.org/docs/latest-v8.x/api/assert.html). I don't think it is available in the version we should be using but correct me if I am wrong. And are you sure it a good idea to avoid using `catch` by any means?

Reply:
Oh if it's not available for node 8 then maybe this is more trouble than it's worth, as we wouldn't want to lose support for it over this. I think the issue you're seeing though is because it needs an `await` before the `assert.rejects`. Maybe one day if there are other reasons to drop support for node 8 we could revisit this, but for now it's not a big deal.

Reply:
I have had the exact issue with `await`. :thinking: The weird thing is that I am on Node.js v8.11.4 and the method exists (if I install `@types/node`) but it behaves in a weird way.

Reply:
Yeah don't worry about it then, we'll leave it as is.